rflx/expression.py

```diff
@@ -1885,13 +1885,13 @@ def simplified(self) -> Expr:
         return Conversion(self.name, self.argument.simplified(), self.type_, self.location)
 
     def ada_expr(self) -> ada.Expr:
-        return ada.Conversion(self.name, self.argument.ada_expr())
+        return ada.Conversion(ada.ID(self.name), self.argument.ada_expr())
 
     def z3expr(self) -> z3.ExprRef:
         raise NotImplementedError
 
     def validate(self, declarations: Mapping[ID, Declaration]) -> None:
-        self.argument.validate(declarations)
+        raise NotImplementedError
```

Comment:
Why is the validation omitted for the converted expression now?

Reply:
It is not covered by any test anymore. I guess this is caused by the changes in the `tests/tls_*_session.rflx` files. As the session validation will be refactored in #395, I don't see any benefit in a deeper investigation here.

Reply:
OK.